### PR TITLE
fix: alpha 4 fixes

### LINKED ISSUE #2165: Alpha 4 cleanup
## Context
After releasing alpha 4, there were a few fixes necessary in the Rafiki behaviour.

## Todos
- [ ] Make fixed-send cross currency quoting work (when exchange rate is not-rounded)
- [ ] Allow setting 0 fixed fee in Admin UI

### PR DESCRIPTION
**Pull request by mkurapov**
## Changes proposed in this pull request

- Fixes BigInt arithmetic issue when creating a cross-currency quote with fees
- Allows to set 0 fixed fee in admin UI
- Give better error for receiver creation when incoming payment is already completed (instead of failing on payment method lookup)
- Increase initial account balance for most frequently used local playground users

## Context

- Fixes #2165 

## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass
- [ ] Postman collection updated

### CODE REVIEW

File: packages/backend/src/open_payments/quote/service.ts [hunk 1]
```diff
@@ -224,8 +224,10 @@ function calculateQuoteAmountsAfterFees(
     const estimatedExchangeRate =
       quote.estimatedExchangeRate || quote.lowEstimatedExchangeRate.valueOf()
 
-    const exchangeAdjustedFees = BigInt(Number(fees) * estimatedExchangeRate)
-    receiveAmountValue = BigInt(receiveAmountValue) - exchangeAdjustedFees
+    const exchangeAdjustedFees = BigInt(
+      Math.ceil(Number(fees) * estimatedExchangeRate)
```
Comment (mkurapov) [inline packages/backend/src/open_payments/quote/service.ts:228]: Main issue was that this wasn't ceil-ed properly, and BigInt couldn't parse

File: packages/backend/src/open_payments/receiver/model.ts [hunk 2]
```diff
@@ -47,6 +43,10 @@ export class Receiver {
       throw new Error('Cannot create receiver from expired incoming payment')
     }
 
+    if (!incomingPayment.methods.length) {
```
Comment (mkurapov) [inline packages/backend/src/open_payments/receiver/model.ts:46]: moved this below the payment completed or expired check, to provide clearer error when creating outgoing payment

File: packages/frontend/app/lib/validate.server.ts [hunk 3]
```diff
@@ -80,7 +80,7 @@ export const setAssetFeeSchema = z.object({
     .bigint({
       invalid_type_error: 'Fixed Fee amount is expected to be a number.'
     })
-    .positive()
+    .nonnegative()
```
Comment (mkurapov) [inline packages/frontend/app/lib/validate.server.ts:83]: we should be able to set 0 fees in for asset

File: packages/backend/src/open_payments/quote/service.test.ts [hunk 4]
```diff
@@ -544,18 +544,20 @@ describe('QuoteService', (): void => {
...
-        ${200n}          | ${0}     | ${0}          | ${100n}                    | ${'no fees'}
-        ${200n}          | ${20}    | ${0}          | ${90n}                     | ${'fixed fee'}
-        ${200n}          | ${0}     | ${200}        | ${99n}                     | ${'basis point fee'}
-        ${200n}          | ${20}    | ${200}        | ${89n}                     | ${'fixed and basis point fee'}
+        debitAmountValue | fixedFee | basisPointFee | exchangeRate | expectedReceiveAmountValue | description
+        ${200n}          | ${0}     | ${0}          | ${0.5}       | ${100n}                    | ${'no fees'}
+        ${200n}          | ${20}    | ${0}          | ${0.5}       | ${90n}                     | ${'fixed fee'}
+        ${200n}          | ${0}     | ${200}        | ${0.5}       | ${99n}                     | ${'basis point fee'}
+        ${200n}          | ${20}    | ${200}        | ${0.5}       | ${89n}                     | ${'fixed and basis point fee'}
+        ${200n}          | ${20}    | ${200}        | ${0.455}     | ${80n}                     | ${'fixed and basis point fee with floating exchange rate'}
```
Comment (mkurapov) [inline packages/backend/src/open_payments/quote/service.test.ts:552]: added test to make sure error is caught

File: packages/backend/src/fee/model.ts [hunk 5]
```diff
@@ -34,6 +34,6 @@ export class Fee extends BaseModel {
     const feePercentage = this.basisPointFee / 10_000
 
     // TODO: bigint/float multiplication
-    return BigInt(Math.floor(Number(principal) * feePercentage)) + this.fixedFee
+    return BigInt(Math.ceil(Number(principal) * feePercentage)) + this.fixedFee
```
Comment (mkurapov) [inline packages/backend/src/fee/model.ts:37]: all fees should be ceil-ed IMO

File: localenv/cloud-nine-wallet/seed.yml [hunk 6]
```diff
@@ -28,12 +28,12 @@ accounts:
   - name: 'Grace Franklin'
     path: accounts/gfranklin
     id: 742ab7cd-1624-4d2e-af6e-e15a71638669
-    initialBalance: 4000
+    initialBalance: 50000
```
Comment (mkurapov) [inline localenv/cloud-nine-wallet/seed.yml:31]: better this than to run into unable to fund outgoing payment issues IMO

**Review by sabineschaller — APPROVED** [review]
Love your explanatory comments!

**Pull request merged**